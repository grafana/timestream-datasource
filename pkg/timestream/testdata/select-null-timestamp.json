{
    "ColumnInfo": [
        {
            "Name": "measure_name",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "VARCHAR",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "_col1",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "TIMESTAMP",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        }
    ],
    "NextToken": null,
    "QueryId": "#QueryId#",
    "QueryStatus": {
        "CumulativeBytesMetered": 10000000,
        "CumulativeBytesScanned": 689699,
        "ProgressPercentage": 27.12738525012893
    },
    "Rows": [
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_end_state",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-04-12 15:08:56.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "network_bytes_out",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:16:49.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "network_bytes_out",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:23:40.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_end_state",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:10:08.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_end_state",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:16:49.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_end_state",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:23:40.000000000",
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_completed",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": true,
                    "RowValue": null,
                    "ScalarValue": null,
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_completed",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": true,
                    "RowValue": null,
                    "ScalarValue": null,
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "task_completed",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": true,
                    "RowValue": null,
                    "ScalarValue": null,
                    "TimeSeriesValue": null
                }
            ]
        },
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "network_bytes_out",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2021-03-21 21:10:08.000000000",
                    "TimeSeriesValue": null
                }
            ]
        }
    ]
}