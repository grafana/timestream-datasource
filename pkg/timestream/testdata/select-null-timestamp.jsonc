//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
//  
//  Frame[0] {
//      "typeVersion": [
//          0,
//          0
//      ]
//  }
//  Name: 
//  Dimensions: 2 Fields by 10 Rows
//  +--------------------+-------------------------------+
//  | Name: measure_name | Name: _col1                   |
//  | Labels:            | Labels:                       |
//  | Type: []*string    | Type: []*time.Time            |
//  +--------------------+-------------------------------+
//  | task_end_state     | 2021-04-12 15:08:56 +0000 UTC |
//  | network_bytes_out  | 2021-03-21 21:16:49 +0000 UTC |
//  | network_bytes_out  | 2021-03-21 21:23:40 +0000 UTC |
//  | task_end_state     | 2021-03-21 21:10:08 +0000 UTC |
//  | task_end_state     | 2021-03-21 21:16:49 +0000 UTC |
//  | task_end_state     | 2021-03-21 21:23:40 +0000 UTC |
//  | task_completed     | null                          |
//  | task_completed     | null                          |
//  | task_completed     | null                          |
//  | network_bytes_out  | 2021-03-21 21:10:08 +0000 UTC |
//  +--------------------+-------------------------------+
//  
//  
//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
{
  "status": 200,
  "frames": [
    {
      "schema": {
        "meta": {
          "typeVersion": [
            0,
            0
          ]
        },
        "fields": [
          {
            "name": "measure_name",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "_col1",
            "type": "time",
            "typeInfo": {
              "frame": "time.Time",
              "nullable": true
            }
          }
        ]
      },
      "data": {
        "values": [
          [
            "task_end_state",
            "network_bytes_out",
            "network_bytes_out",
            "task_end_state",
            "task_end_state",
            "task_end_state",
            "task_completed",
            "task_completed",
            "task_completed",
            "network_bytes_out"
          ],
          [
            1618240136000,
            1616361409000,
            1616361820000,
            1616361008000,
            1616361409000,
            1616361820000,
            null,
            null,
            null,
            1616361008000
          ]
        ]
      }
    }
  ]
}