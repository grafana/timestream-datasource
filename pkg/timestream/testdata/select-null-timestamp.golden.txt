ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ

Frame[0] {}
Name: 
Dimensions: 2 Fields by 10 Rows
+--------------------+-------------------------------+
| Name: measure_name | Name: _col1                   |
| Labels:            | Labels:                       |
| Type: []*string    | Type: []time.Time             |
+--------------------+-------------------------------+
| task_end_state     | 2021-04-12 15:08:56 +0000 UTC |
| network_bytes_out  | 2021-03-21 21:16:49 +0000 UTC |
| network_bytes_out  | 2021-03-21 21:23:40 +0000 UTC |
| task_end_state     | 2021-03-21 21:10:08 +0000 UTC |
| task_end_state     | 2021-03-21 21:16:49 +0000 UTC |
| task_end_state     | 2021-03-21 21:23:40 +0000 UTC |
| task_completed     | 0001-01-01 00:00:00 +0000 UTC |
| task_completed     | 0001-01-01 00:00:00 +0000 UTC |
| task_completed     | 0001-01-01 00:00:00 +0000 UTC |
| network_bytes_out  | 2021-03-21 21:10:08 +0000 UTC |
+--------------------+-------------------------------+


====== TEST DATA RESPONSE (arrow base64) ======
FRAME=QVJST1cxAAD/////sAEAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEEAAoADAAAAAgABAAKAAAACAAAAHQAAAADAAAATAAAACgAAAAEAAAA7P7//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAAAM////CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAACz///8IAAAADAAAAAIAAAB7fQAABAAAAG1ldGEAAAAAAgAAAJAAAAAYAAAAAAASABgAFAAAABMADAAAAAgABAASAAAAFAAAADwAAABEAAAAAAAACkQAAAABAAAABAAAAIz///8IAAAAEAAAAAUAAABfY29sMQAAAAQAAABuYW1lAAAAAAAAAAAAAAYACAAGAAYAAAAAAAMABQAAAF9jb2wxABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAATAAAAFAAAAAAAAUBTAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABgAAAAMAAAAbWVhc3VyZV9uYW1lAAAAAAQAAABuYW1lAAAAAAAAAAAEAAQABAAAAAwAAABtZWFzdXJlX25hbWUAAAAAAAAAAP/////IAAAAFAAAAAAAAAAMABYAFAATAAwABAAMAAAAGAEAAAAAAAAUAAAAAAAAAwQACgAYAAwACAAEAAoAAAAUAAAAaAAAAAoAAAAAAAAAAAAAAAUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsAAAAAAAAADAAAAAAAAAAlQAAAAAAAADIAAAAAAAAAAAAAAAAAAAAyAAAAAAAAABQAAAAAAAAAAAAAAACAAAACgAAAAAAAAAAAAAAAAAAAAoAAAAAAAAAAAAAAAAAAAAAAAAADgAAAB8AAAAwAAAAPgAAAEwAAABaAAAAaAAAAHYAAACEAAAAlQAAAAAAAAB0YXNrX2VuZF9zdGF0ZW5ldHdvcmtfYnl0ZXNfb3V0bmV0d29ya19ieXRlc19vdXR0YXNrX2VuZF9zdGF0ZXRhc2tfZW5kX3N0YXRldGFza19lbmRfc3RhdGV0YXNrX2NvbXBsZXRlZHRhc2tfY29tcGxldGVkdGFza19jb21wbGV0ZWRuZXR3b3JrX2J5dGVzX291dAAAAABQj3PUJHUWAEoYPCN4bhYAmJrtgnhuFgDgod7Fd24WAEoYPCN4bhYAmJrtgnhuFgAAGj3rA7KhAAAaPesDsqEAABo96wOyoQDgod7Fd24WEAAAAAwAFAASAAwACAAEAAwAAAAQAAAALAAAADgAAAAAAAQAAQAAAMABAAAAAAAA0AAAAAAAAAAYAQAAAAAAAAAAAAAAAAAAAAAKAAwAAAAIAAQACgAAAAgAAAB0AAAAAwAAAEwAAAAoAAAABAAAAOz+//8IAAAADAAAAAAAAAAAAAAABQAAAHJlZklkAAAADP///wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAAs////CAAAAAwAAAACAAAAe30AAAQAAABtZXRhAAAAAAIAAACQAAAAGAAAAAAAEgAYABQAAAATAAwAAAAIAAQAEgAAABQAAAA8AAAARAAAAAAAAApEAAAAAQAAAAQAAACM////CAAAABAAAAAFAAAAX2NvbDEAAAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAUAAABfY29sMQASABgAFAATABIADAAAAAgABAASAAAAFAAAAEwAAABQAAAAAAAFAUwAAAABAAAADAAAAAgADAAIAAQACAAAAAgAAAAYAAAADAAAAG1lYXN1cmVfbmFtZQAAAAAEAAAAbmFtZQAAAAAAAAAABAAEAAQAAAAMAAAAbWVhc3VyZV9uYW1lAAAAANgBAABBUlJPVzE=
